src/components/ModelingSidebar/ModelingPanes/MemoryPane.tsx

```diff
@@ -93,13 +93,13 @@ export const processMemory = (programMemory: ProgramMemory) => {
       // @ts-ignore
       (val.type !== 'Function' && val.type !== 'UserVal')
     ) {
-      const sg = sketchFromKclValue(val, key)
+      const sk = sketchFromKclValueOptional(val, key)
       if (val.type === 'Solid') {
         processedMemory[key] = val.value.map(({ ...rest }: ExtrudeSurface) => {
           return rest
         })
-      } else if (!err(sg)) {
-        processedMemory[key] = sg.paths.map(({ __geoMeta, ...rest }: Path) => {
+      } else if (!(sk instanceof Reason)) {
+        processedMemory[key] = sk.paths.map(({ __geoMeta, ...rest }: Path) => {
```

Comment:
If you can assure me this is actually not an error, I'll approve this to add to our toolbox.

Reply:
What do you mean? The return type of `sketchFromKclValueOptional()` cannot return an error.

Reply:
@lf94, does this answer your question?

Reply:
Whoops sorry, yea I'll approve it!